pubpub/platform
# Run emit event task when pub enters stage

This PR creates a basic scaffolding that will become our eventemitter. It uses a database trigger and  a [graphile worker sql function](https://worker.graphile.org/docs/sql-add-job#graphile_workeradd_job) to create a job whenever a row is changed in the `PubsInStages` table, which will allow us to run any action instances that subscribe to the `"pub-entered"` event. This doesn't actually look up and run the actions yet, since @tefkah is working on code related to that aspect. It also doesn't yet include a way for action instances to subscribe to events (or add rules/triggers or whatever we'll call them), which feels like a separate issue to me: #313.

## Issue(s) Resolved
#246

## Test Plan
1. Run `pnpm migrate-dev`
2. Submit a pub if you don't have any
3. Observe that the worker task logs the database change in your console
4. Return to the stages page and move your pub(s) around. Each change should be logged by the worker

## Screenshots (if applicable)

## Optional

### Notes/Context/Gotchas
I upgraded graphile worker, which might throw some errors for you. I'd just truncate the private jobs table if you have issues, although I did not. We should be careful about deploying this to prod though!

### Supporting Docs
